racket/collects/pkg/strip.rkt

```diff
@@ -503,3 +500,25 @@
                     which
                     dir)
             (current-continuation-marks)))))
+
+(define (call-with-output-file/writable pth proc)
+  ;; In case `pth` was copied from a file without the user-write-bit set,
+  ;; explicitly make it writable while we overwrite it.
+  (define (run)
+    (call-with-output-file* pth
+      #:exists 'truncate/replace
+      proc))
+  (cond
+    [(file-exists? pth)
+     (define old-mode
+       (file-or-directory-permissions pth 'bits))
+     (define new-mode
+       (bitwise-ior old-mode user-write-bit))
```

Comment:
I think this won't work on Windows, where the only thing you can control via `file-or-directory-permissions` is whether writing is allowed for everyone (i.e., no user vs. group vs. other distinction).

So, `(if (eq? (system-type) 'windows) (bitwise-ior user-write-bit group-write-bit other-write-bit) user-write-bit)`? 

Also, maybe it would be better to try to change permissions only if `user-write-bit` is not already set.

Reply:
Oh, I bet this is also the reason why tests for https://github.com/racket/racket/pull/4126 failed on Windows. I had seen in the documentation that Windows doesn't have a user/group/other distinction, but I guess I must have thought the non-user bits were just ignored or managed automatically somehow: I didn't realize the user of `file-or-directory-permissions` was responsible for keeping the bits consistent.

From the linked issues there, it seems like I got it wrong in the same way in https://github.com/racket/racket/commit/55b6cbdca1f36a4f37bab1519c1b658717d3cad2: I'll try to fix that, too.